wazo_confd/plugins/phone_number/validator.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from xivo_dao.helpers import errors
+from xivo_dao.resources.phone_number import dao
+
+from wazo_confd.helpers.validator import (
+    ValidationGroup,
+    Validator,
+    UniqueInTenantValidator,
+)
+
+
+class OnlyOneMainByTenantValidator(Validator):
+    def __init__(self, dao):
+        self.dao = dao
+
+    def validate(self, phone_number):
+        if not phone_number.main:
+            return
+
+        existing = self.dao.find_by(main=True, tenant_uuid=phone_number.tenant_uuid)
+        if existing and existing.uuid != phone_number.uuid:
+            raise errors.resource_exists(
+                'PhoneNumber',
+                main=True,
+                tenant_uuid=str(phone_number.tenant_uuid),
+            )
```

Comment:
why do we need to add the overhead of an additional query if we have a database constraint?

Reply:
It's the pattern in wazo-confd where we check first and hope nothing changes after the check. I would prefer handling the IntegrityError but that would be a first in wazo-confd and would need some discussion to introduce this mechanic in the "framework" which has almost no logic other than the marshmallow schema and the sqlalchemy model